autotest.py

```diff
@@ -63,8 +63,6 @@ def run_autotest():
         )
         return 0
 
-    copy_files_to_temp_directory(args, parameters)
```

Comment:
Was this intentional?

If this is removed, there is no reference to the `copy_files_to_temp_directory` function at all in `autotest.py`

Reply:
It was intentional. However, you are right to comment on the ambiguity. I have discovered two issues with this change. One is easily fixed (generate_expected_output needs to run after copy_files_to_temp_directory). The other, however, involves a kind of circular dependency where the run_tests function (with the new changes) relies on upload_results having already run, and upload_results depends on run_tests having run...I have a solution for this, however. Will be pushed soon.